appcues/src/main/java/com/appcues/SessionMonitor.kt

```diff
@@ -48,7 +48,10 @@ internal class SessionMonitor(
     }
 
     fun reset() {
+        // reset storage before calling track so it can use updated storage information
+        storage.reset()
         analyticsTracker.track(AnalyticsEvent.SessionReset, null, false)
+        // set sessionId to null AFTER the track or else it wont track anything as checkSession will return false
```

Comment:
I _think_ this might be incorrect.  If you reset before the SessionReset, then that analytic won't actually track correctly - since no user.  I thought the analytic track had to happen before the values get wiped out.